### PR TITLE
Feature/random controversy

### LINKED ISSUE #5: As a user, I should be able to enter the name of a public figure into the search bar and find a wikipedia article that displays their associated controversies.

### PR DESCRIPTION
**Pull request by MaryBruff**
What I did:

⭐ Move search to custom hook folder 
> ✨ Which abstracts the search functionality into three separate functions that can be pulled into a component.

⭐ Added an array of some of the search inputs to display as random controversies on page load. 
>  ❗ These persist when another search is entered so we will have to decide what to do with these once Login is incorporated. 

Before Search:
image

After looking something up:
image

⭐ Also removed test files in components because we don't need them since we are testing with Cypress! 

Did this break anything?

- [X] No
- [ ]  Yes

Type of change

- [X] New feature (non-breaking change which adds functionality)
- [ ]  Styling 
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

Checklist:

- [ ]  If this code needs to be tested, all tests are passing
- [X]  The code runs locally
- [ ]  There are comments where clarification/ organization is needed
- [X]  The code is DRY. If it's not, I tried my best
- [X]  I reviewed my code before pushing

What's next?
⚠️ Finish implementing Auth and Login functionality 
🚧 Fix Router after Auth and Login

This PR closes #5

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+import React, { useEffect } from 'react';
 import { Routes, Route, Navigate } from 'react-router-dom';
 import WikipediaSearch from '../Search/Search.js';
 import './App.css';
+import Card from '../Card/Card.js';
+import useSearchResults from '../hooks/useSearchResults.js';
+
 
 function App() {
+  const randomSearchInputs = ['4chan', 'Titan (Submersible)', 'Billy Mitchell (gamer)', 'You Showed Me', 'Ezra Miller', 'Russel Brand', 'Bernie Madoff', 'Amy Winehouse']
```
Comment (Jesuitman) [inline src/App/App.js:10]: woohoo! love you using the gamer

File: src/App/App.js [hunk 2]
```diff
@@ -12,9 +28,20 @@ function App() {
       </header>
       <Routes>
         <Route path='/' element={<WikipediaSearch />} />
-        <Route path='/main' element={<Navigate to='/' />} />
+        {/* <Route path='/main' element={<Navigate to='/' />} /> */}
         <Route path="article/:id" element={<WikipediaSearch />} />
       </Routes>
+      <footer className='footer-card'>
+        <h2 className='footer-text'>Random Controversy</h2>
+        {controversies[0] && <h2 id='resultName'>{initialResults.title}</h2>}
```
Comment (Jesuitman) [inline src/App/App.js:36]: excellent use of randomization to load a different one every time!

File: src/Card/Card.js [hunk 3]
```diff
@@ -13,7 +13,6 @@ const Card = ({ snippet }) => {
   const handleTitleClick = () => {
     setShowFullContent(!showFullContent);
   };
-
```
Comment (Jesuitman) [inline src/Card/Card.js:16]: yeah remove that space

File: src/Search/Search.js [hunk 4]
```diff
@@ -2,133 +2,35 @@ import React, { useState, useEffect } from 'react';
...
   
-  async function displaySearchResults(searchTerm) {
-    const url = 'https://en.wikipedia.org/w/api.php?';
-    let params = {
-      action: 'query',
-      list: 'search',
-      format: 'json',
-      origin: '*',
-      srlimit: 20,
-      srsearch: searchTerm,
```
Comment (Jesuitman) [inline src/Search/Search.js:33]: its awesome you were able to move these to a different space and then still be able to call on them, that makes things look super streamlined

File: src/hooks/useSearchResults.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        throw new Error('Network response was not ok');
+      }
+      return response.json();
+    })
+    .then((data) => {
+      // Sets the state of the initial results to the first result of the search (i.e. Rush Limbaugh)
+      setInitialResults(data.query.search[0]);
+    })
+    .catch((error) => {
+      console.error('There was a problem with the fetch operation:', error);
```
Comment (Jesuitman) [inline src/hooks/useSearchResults.js:32]: great use of a hooks file to actually get this isolated and shrink file size

**Review by Jesuitman — APPROVED** [review]

**Review by JoeyStewart — APPROVED** [review]

**Pull request merged**